src/dsl.rs

```diff
@@ -33,10 +33,18 @@ impl<F, TraceArgs, StepArgs> CircuitContext<F, TraceArgs, StepArgs> {
         Queriable::Forward(self.sc.add_forward(name, phase), false)
     }
 
+    /// Adds a shared signal to the circuit with a name string and zero rotation and returns a
+    /// `Queriable` instance representing the added shared signal.
     pub fn shared(&mut self, name: &str) -> Queriable<F> {
         Queriable::Shared(self.sc.add_shared(name, 0), 0)
     }
 
+    /// Adds a shared signal to the circuit with a name string and a specified phase and returns a
+    /// `Queriable` instance representing the added shared signal.
+    pub fn shared_with_phase(&mut self, name: &str, phase: usize) -> Queriable<F> {
+        Queriable::Shared(self.sc.add_shared(name, phase), 0)
+    }
```

Comment:
Made these changes in this PR, so that I only need to merge in Part 3.

Reply:
I don't understand this sentence. Why is this showing this up if this is from part 1.

Reply:
Apologies! I merged part 1 to main before fixing this so that I can merge main in part 2 rather than asking you to approve changes in part 1 again. This way it's quicker. However, I think you are right that for code organization reason it's the best to keep it in part 1.